src/classes/PS_CommitmentRequest.cls

```diff
@@ -286,6 +260,45 @@ public inherited sharing class PS_CommitmentRequest {
         return commitment;
     }
 
+    /***
+    * @description Retrieves Contact data specified as the donor on the Recurring Donation record
+    * that is not yet saved into database
+    * @param contactId Contact Id
+    * @return Contact
+    */
+    private Contact getContact(Id contactId) {
+        List<Contact> contacts = new List<Contact>([
+            SELECT
+                AccountId, FirstName, LastName
+            FROM Contact
+            WHERE Id = :contactId
+            LIMIT 1
+        ]);
+        
+        return contacts == null || contacts.isEmpty()
```

Comment:
Does not need the null check because the query will always return a List even if no record is found.
Same with `getAccount()` method below